client_test.go

```diff
@@ -177,6 +213,108 @@ func TestProvisionResource(t *testing.T) {
 		gm.Expect(async).To(gm.BeTrue(), "Result on 202 should be async")
 		gm.Expect(message).To(gm.Equal("please wait"))
 	})
+
+	t.Run("400 bad request valid response", func(t *testing.T) {
+		gm.RegisterTestingT(t)
+		srv := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			rw.Header().Add("Content-Type", "application/json")
+			rw.WriteHeader(http.StatusBadRequest)
+			rw.Write([]byte(`{"message":"i dont get ya"}`))
+		}))
+		defer srv.Close()
+
+		_, _, err := callProvision(srv.URL)
+
+		gm.Expect(err).To(gm.HaveOccurred())
+		e := err.(*Error)
+
+		gm.Expect(e.Type).To(gm.Equal(errors.BadRequestError))
+		gm.Expect(e.Message).To(gm.Equal("i dont get ya"))
+	})
+
+	t.Run("400 bad request with invalid content-type", func(t *testing.T) {
+		gm.RegisterTestingT(t)
+		srv := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			rw.Header().Add("Content-Type", "text/html")
+			rw.WriteHeader(http.StatusBadRequest)
+			rw.Write([]byte(`{"message":"i dont get ya"}`))
+		}))
+		defer srv.Close()
+
+		_, _, err := callProvision(srv.URL)
+
+		gm.Expect(err).To(gm.HaveOccurred())
+		gm.Expect(err.Error()).To(gm.Equal("no consumer: \"text/html\""))
```

Comment:
You can combine have occurred and string matching as:
```go
gm.Expect(err).To(gm.MatchError(`no consumer: "text/html"`))
```
you can also use MatchError with a type that implements the error interface, in which case the two will be deepequaled. could be useful for the other error checking:

```go
gm.Expect(err).To(gm.MatchError(manifold.NewError(errors.BadRequestError, "i dont get ya"))
```